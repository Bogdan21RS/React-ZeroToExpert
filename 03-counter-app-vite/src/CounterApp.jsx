import { useState } from "react";
import PropTypes from "prop-types";

export default function CounterApp({ value }) {
  const [counter, setCounter] = useState(value);

  function handleAdd() {
    // setCounter( (counter) => counter + 1)
    setCounter(counter + 1);
  }

  function handleSubstract() {
    setCounter(counter - 1);
  }

  function handleReset() {
    setCounter(value);
  }

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>

      <button onClick={handleAdd}>+1</button>
      <button onClick={handleSubstract}>-1</button>
      <button onClick={handleReset}>reset</button>
    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
