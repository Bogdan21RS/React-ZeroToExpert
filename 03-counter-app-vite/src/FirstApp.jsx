export default function FirstApp() {
  return (
    <>
      <h1>First App</h1>
      <p>This is a paragraph</p>
    </>
  );
}
