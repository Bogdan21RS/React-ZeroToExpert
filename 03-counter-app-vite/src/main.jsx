import React from "react";
import ReactDom from "react-dom/client";
import HelloWorldApp from "./HelloWorldApp";
// import CounterApp from "./CounterApp";
import FirstApp from "./firstApp";
import "./styles.css";

ReactDom.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <HelloWorldApp /> */}
    {/* <CounterApp value={10} /> */}
    <FirstApp title="hi" />
  </React.StrictMode>
);
