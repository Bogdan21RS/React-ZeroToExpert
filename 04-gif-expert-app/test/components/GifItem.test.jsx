import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("GifItem test", () => {
  const title = "Saitama";
  const url = "http://localhost/";

  test("should match with the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should show the image with the given alt text ant title", () => {
    render(<GifItem title={title} url={url} />);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("should show the title in the component", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
