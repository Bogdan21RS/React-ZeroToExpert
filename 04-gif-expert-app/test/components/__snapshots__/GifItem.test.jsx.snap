// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem test should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://localhost/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
